### PR TITLE
(#125) Token Support

### LINKED ISSUE #125: Generalised Bridge - Token Support
Update the Validator node in such a way that:
- Crypto Transfer Watcher is refactored to Transfer Watcher - watching not only hbar transfers but token transfers as-well
- Crypto Transfer Handler is refactored to Transfer handler - handling not only hbar transfers but token transfers as-well
- Add new column in the Transaction record in DB for specifying the ERC20 Contract address
- Update the signed message to include the token address and publish it to HCS
- Update the Consensus Message handler to parse the new signature format and be able to support transfer messages for tokens as-well.
- Add new column in HCS Messages records in DB for populating storing the ERC20 contract address

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
**Detailed description**:
This PR modifies the already implemented logic and extends it to support "listening" to token transfers.

**Not Tested!**
This implementation is still not fully tested, because a mirror-node api problem is still being resolved.

**Which issue(s) this PR fixes**:
Fixes #125 

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: proto/transfer_message.proto [hunk 1]
```diff
@@ -11,4 +11,5 @@ message TransferMessage {
   string amount = 3;
   string fee = 4;
   string gasPriceGwei = 5;
+  string asset = 6;
```
Comment (Daniel-K-Ivanov) [inline proto/transfer_message.proto:14]: We should add erc20 address as-well

File: app/process/watcher/transfer/watcher.go [hunk 2]
```diff
@@ -127,19 +130,29 @@ func (ctw Watcher) beginWatching(q *queue.Queue) {
 
 func (ctw Watcher) processTransaction(tx mirror_node.Transaction, q *queue.Queue) {
 	ctw.logger.Infof("New Transaction with ID: [%s]", tx.TransactionID)
-	amount, err := tx.GetIncomingAmountFor(ctw.accountID.String())
+	amount, asset, err := tx.GetIncomingTokenAmountFor(ctw.accountID.String())
+	if err != nil {
+		ctw.logger.Errorf("Could not extract incoming token amount for TX [%s]. Error: [%s]", tx.TransactionID, err)
```
[the remark this answers is not in the document]
Comment (georgiyazovaliiski) [reply app/process/watcher/transfer/watcher.go:135]: I dropped it on purpose. If we fail to get Token Amount -> Continue to retrieving Hbar Amount.

**Review by failfmi — APPROVED** [review]

File: app/persistence/transaction/transaction.go [hunk 3]
```diff
@@ -83,6 +83,7 @@ type Transaction struct {
 	EthTxStatus           string
 	EthHash               string
 	GasPriceGwei          string
+	Asset			      string
```
Comment (failfmi) [inline app/persistence/transaction/transaction.go:86]: Why is this showing it with such indentation

Comment (georgiyazovaliiski) [reply app/persistence/transaction/transaction.go:86]: This is the first time I am seeing it... The IDE displayed it as it should be.

**Review by failfmi — APPROVED** [review]

File: app/clients/hedera/mirror-node/transaction.go [hunk 4]
```diff
@@ -37,11 +37,14 @@ type (
...
 		TransactionID        string `json:"transaction_id"`
 		Transfers            []Transfer
+		TokenTransfers       []Transfer
 	}
 	// Transfer struct used by the Hedera Mirror node REST API
 	Transfer struct {
 		Account string `json:"account"`
 		Amount  int64  `json:"amount"`
+		// When retrieving ordinary hbar transfers, this field does not get populated
+		Asset string `json:"asset"`
```
Comment (Daniel-K-Ivanov) [inline app/clients/hedera/mirror-node/transaction.go:47]: nit: not formatted

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**